core/client/src/light/fetcher.rs

```diff
@@ -135,6 +144,8 @@ pub trait Fetcher<Block: BlockT>: Send + Sync {
 	type RemoteCallResult: IntoFuture<Item=Vec<u8>, Error=ClientError>;
 	/// Remote changes result future.
 	type RemoteChangesResult: IntoFuture<Item=Vec<(NumberFor<Block>, u32)>, Error=ClientError>;
+	/// Remote block body result future.
+	type RemoteBodyResult: IntoFuture<Item = Option<Vec<Block::Extrinsic>>, Error = ClientError>;
```

Comment:
I think you should use `Vec<Block::Extrinsic>` instead of `Option<Vec<Block::Extrinsic>>` here - the block is header + body even though body could be empty.

Reply:
Yeah I agree, I choose `Option<Vec<Block::Extrinsic>>` because of https://github.com/paritytech/substrate/blob/master/core/client/src/blockchain.rs#L78 but changed now.